### PR TITLE
doc: differentiate SmartOS and SunOS support

### PR DESCRIPTION
**Pull request by cjihrig**
Refs: https://github.com/libuv/libuv/issues/1458
Refs: https://github.com/libuv/libuv/pull/991

This would also require an owner to rename the existing sunos team.

### CODE REVIEW

File: SUPPORTED_PLATFORMS.md [hunk 1]
```diff
@@ -9,9 +9,10 @@
 | AIX | Tier 2 | >= 6 | Maintainers: @libuv/aix |
 | z/OS | Tier 2 | >= V2R2 | Maintainers: @libuv/zos |
 | Linux with musl | Tier 2 | musl >= 1.0 | |
-| SunOS | Tier 2 | Solaris 121 and later | Maintainers: @libuv/sunos |
+| SmartOS | Tier 2 | >= 14.4 | Maintainers: @libuv/smartos |
 | Android | Tier 3 | NDK >= r15b | |
 | MinGW | Tier 3 | MinGW32 and MinGW-w64 | |
+| SunOS | Tier 3 | Solaris 121 and later | |
```
Comment (saghul) [inline SUPPORTED_PLATFORMS.md]: Doesn't this have a maintainer now?

Comment (cjihrig) [inline SUPPORTED_PLATFORMS.md]: No. I moved it to tier 3 because, AFAIK, SunOS never had a maintainer - just @misterdjules, but that was by mistake.

Comment (saghul) [inline SUPPORTED_PLATFORMS.md]: Ack.

**Comment by cjihrig** [comment]
Thanks for the reviews. Friendly reminder to @saghul to rename the sunos team.

**Comment by cjihrig** [comment]
@saghul can you rename @libuv/sunos to @libuv/smartos (refs https://github.com/libuv/libuv/pull/1514#issuecomment-327210190)

**Comment by saghul** [comment]
Done!

**Pull request merged**